polars/polars-core/src/frame/mod.rs

```diff
@@ -3346,29 +3347,51 @@ impl DataFrame {
                             // groups are in bounds
                             unsafe { self._take_unchecked_slice(&group, false) }
                         })
+                        .map(|group| {
+                            if include_key {
+                                group
+                            } else {
+                                group.drop_many(cols)
+                            }
+                        })
                         .collect())
                 }
                 GroupsProxy::Slice { groups, .. } => Ok(groups
                     .into_par_iter()
                     .map(|[first, len]| self.slice(first as i64, len as usize))
+                    .map(|group| {
+                        if include_key {
+                            group
+                        } else {
+                            group.drop_many(cols)
```

Comment:
Can you use `drop_many_amortized`. That saves an allocation per group. See #9814

Reply:
Thanks @ritchie46--I'll do that once that gets merged into main (`drop_many_ amortized` isnt' available yet).